### PR TITLE
Fix fatal errors in when Sidebars component is not initialized

### PR DESCRIPTION
**Pull request by benoitchantre**
## Description

Addresses issue #449.

Adds some checks to prevent fatal errors when the Sidebars component is not initialized.

## List of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:
- [x] This pull request relates to a ticket.
- [x] This code is tested.
- [ ] This change has been added to CHANGELOG.md
- [x] I want my code added to WP Rig.

### CODE REVIEW

File: inc/Image_Sizes/Component.php [hunk 1]
```diff
@@ -51,7 +50,8 @@ public function filter_content_image_sizes_attr( string $sizes, array $size ) :
 			$sizes = '100vw';
 		}
 
-		if ( wp_rig()->is_primary_sidebar_active() ) {
+		if ( method_exists( 'wp_rig', 'is_primary_sidebar_active' )
```
Comment (felixarntz) [inline inc/Image_Sizes/Component.php]: I think this should be `method_exists( wp_rig(), 'is_primary_sidebar_active' )` right?

File: inc/Image_Sizes/Component.php [hunk 2]
```diff
@@ -85,7 +85,8 @@ public function filter_header_image_tag( string $html, $header, array $attr ) :
 	public function filter_post_thumbnail_sizes_attr( array $attr, WP_Post $attachment, $size ) : array {
 		$attr['sizes'] = '100vw';
 
-		if ( wp_rig()->is_primary_sidebar_active() ) {
+		if ( method_exists( 'wp_rig', 'is_primary_sidebar_active' )
```
Comment (felixarntz) [inline inc/Image_Sizes/Component.php]: Same here.

File: inc/Styles/Component.php [hunk 3]
```diff
@@ -312,6 +300,26 @@ protected function get_css_files() : array {
 			),
 		);
 
+		if ( method_exists( 'wp_rig', 'is_primary_sidebar_active' ) ) {
```
Comment (felixarntz) [inline inc/Styles/Component.php]: Same here.

File: sidebar.php [hunk 4]
```diff
@@ -9,7 +9,8 @@
 
 namespace WP_Rig\WP_Rig;
 
-if ( ! wp_rig()->is_primary_sidebar_active() ) {
+if ( ! method_exists( 'wp_rig', 'is_primary_sidebar_active' )
```
Comment (felixarntz) [inline sidebar.php]: Same here.

File: inc/Styles/Component.php [hunk 5]
```diff
@@ -291,18 +291,6 @@ protected function get_css_files() : array {
...
-				'preload_callback' => function() {
-					return wp_rig()->is_primary_sidebar_active();
-				},
-			),
-			'wp-rig-widgets'    => array(
-				'file'             => 'widgets.min.css',
-				'preload_callback' => function() {
-					return wp_rig()->is_primary_sidebar_active();
-				},
-			),
```
Comment (felixarntz) [inline inc/Styles/Component.php]: Could we possibly just keep that bit and add the `method_exists()` checks inside the `preload_callback`? Or is there something I'm missing that wouldn't work then?

Comment (benoitchantre) [reply inc/Styles/Component.php]: If we only check inside the `preload_callback`, the styles will be enqueued when peloading styles is disabled. We don't need these files when there's no Sidebar component.

File: inc/Image_Sizes/Component.php (hunk 1 above)
Comment (benoitchantre) [reply inc/Image_Sizes/Component.php]: It seems I have not tested my code properly: it doesn't work 🤦‍♂  
`method_exists( wp_rig(), 'is_primary_sidebar_active' )` doesn't work either.
`method_exists( 'WP_Rig\WP_Rig\Sidebars\Component', 'is_primary_sidebar_active' )` works, but only if the component has been completely removed (doesn't work if the component is simply not instanciated). I'll need some help to find the right solution.

**Comment by benoitchantre** [comment]
@felixarntz PR is updated. I have added `template_tag_exists` method in `Template_Tags` class and used it instead of `method_exists`.

**Comment by ataylorme** [comment]
@bamadesigner raised a good point in office hours that we haven't had a discussion on being able to remove components. The current codebase assumes all components will exist/be loaded.

**Comment by ataylorme** [comment]
My opinion is that deleting unused components is not necessary. It will leave a bit more PHP but mostly actions/filters not being used if the corresponding feature isn't active (e.g. AMP or JetPack plugins not installed/activated).

In the case of the sidebar, you could simply keep the sidebar inactive and it won't render.

**Comment by bamadesigner** [comment]
Hi @benoitchantre! Thanks for the PR and for joining our Office Hours earlier to discuss. Our core maintainers had a chat and decided we definitely want to explore a world where its easier in WP Rig to add/remove components but we want to give it due diligence so we created a separate issue and were going to work on this functionality in a future release.

You can view the issue here, if you'd like to comment or follow along: https://github.com/wprig/wprig/issues/515

Thanks again!

**Comment by benoitchantre** [comment]
No problem, thank you for your comment @bamadesigner.

**Pull request closed without merging**